hostthedocs/filekeeper.py

```diff
@@ -55,9 +56,15 @@ def unpack_project(zippath, proj_metadata, docfiles_dir):
     with open(descrpath, 'w') as f:
         f.write(proj_metadata.get('description', DEFAULT_PROJECT_DESCRIPTION))
 
-    zf = zipfile.ZipFile(zippath)
     # This is insecure, we are only accepting things from trusted sources.
-    zf.extractall(verdir)
+    if fmt in ['tar', 'tar.gz', 'tar.bz2']:
+        with tarfile.open(fileobj=file_path_stream, mode='r:*') as tf:
+            tf.extractall()
+        if not isinstance(file_path_stream, str):
```

Comment:
Should this be `hasattr(file_path_stream, 'close') rather than checking type? More duck-typed.